debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -314,6 +327,71 @@ public boolean exists() {
         return exists;
     }
 
+    private void checkTopicSettings(String topicName) {
+        final Thread t = new Thread((Runnable) () -> {
```

Comment:
Could you use `ExecutorService` instead? It should be the preferred way for async task execution.

Reply:
@gunnarmorling I tend to disagree here - the thread is fire and forget. If we want to do the same with the Executor we would need to provide a `ThreadFactory` that does exactly the same - create a thread and set it to daemon and then we should close it to ensure a smmetry in the code. I believe this is a more pragmatic solution given its intent

Reply:
The point on thread factory is a good one actually, as it ensures right naming of the thread. See `Threads.newSingleThreadExecutor()`.

Reply:
@gunnarmorling `ThreadFactory` ensure naming only if `setName` is explictly called on the created thread which was done even before. Anyway I've added new commit with rewrite to executors API.